### ISSUE TITLE
[HOLD for payment 2023-10-10] [HOLD for payment 2023-10-09] [$500] Chat - Error appears on click of notification and does not redirect to page

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#28328]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. From userA, send message to userB
4. From userB, click on the notification

## Expected Result:
It  should redirect to userA chat

## Actual Result:
Nothing happens on click of notification and console error appears (_ROUTES__WEBPACK_IMPORTED_MODULE_13__.default.getReportRoute is not a function)

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari -requested 
- [x] MacOS / Desktop

**Version Number:** Dev 1.3.74-2
**Reproducible in staging?:** n
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/c41fbd6d-a4a1-43b2-839d-0f82f400042a

https://github.com/Expensify/App/assets/93399543/d50a40de-ef1e-44fd-a8e4-af4dfb76893b

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1695709767898199

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d1b259b2d9260fc0</li>
        <li>Upwork Job ID: 1707070231351926784</li>
        <li>Last Price Increase: 2023-09-27</li>
<li>Automatic offers: </li>
<ul>
<li>s77rt | Reviewer | 26983371</li>
<li>gadhiyamanan | Reporter | 26983373</li>
</ul></li>
    </ul>
</details>

**Comment by redpanda-bit** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Error shows in console after tapping notification and does not navigate as expected.
### What is the root cause of that problem?
The notification is using a function that is no longer there.
### What changes do you think we should make in order to solve the problem?

The function was moved from `ROUTES.REPORT_WITH_ID.getRoute(reportID)` to`ROUTES.getReportRoute(reportID)`

https://github.com/Expensify/App/blob/13b229116163f13e4cda7d322f1f597744b067e9/src/libs/actions/Report.js#L1652-L1656

### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by cubuspl42** [comment]
@redpanda-bit 

> The function was moved from ROUTES.REPORT_WITH_ID.getRoute(reportID) toROUTES.getReportRoute(reportID)

I'm having trouble understanding what's the proposed solution. Could you try rephrasing it?

**Comment by redpanda-bit** [comment]
TL;DR: The function `ROUTES.getReportRoute(reportID)` no longer exists and the function calls to this function need to be updated to `ROUTES.REPORT_WITH_ID.getRoute(reportID)`

Thanks for asking for clarification!

It's pretty much the same as @Charan-hs 's proposal. I just had the changes (`from` -> `to`) backwards because of rushing to be the first proposal. I think both proposals achieve the exact same result.

**Comment by s77rt** [comment]
@redpanda-bit's proposal looks good to me. Please raise a PR asap

**Comment by redpanda-bit** [comment]
On it @s77rt.

**Comment by situchan** [comment]
This is regression from https://github.com/Expensify/App/pull/27299, not from https://github.com/Expensify/App/pull/28050.
Especially this change:
https://github.com/Expensify/App/pull/27299/files#diff-8afe5b71ee0436c21364148c86dadd640f2bff3e3d916addbb1f1f6f7e5b6a43R1654

**Comment by redpanda-bit** [comment]
Going through PR checklist still.

**Comment by redpanda-bit** [comment]
Is anyone available to help me test the platforms I am missing?

- [ ] I ran the tests on **all platforms** & verified they passed on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop

Or is it ok this time since the change is only a one liner?

**Comment by s77rt** [comment]
@redpanda-bit Please raise the PR and tag me

**Comment by mountiny** [comment]
Taking the management of this one over.

**Comment by mountiny** [comment]
This is fixed in staging. Flat price for this one.

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: https://github.com/Expensify/App/pull/27299
- The offending PR has been commented on: https://github.com/Expensify/App/pull/27299#discussion_r1343209391
- A discussion in #expensify-bugs has been started: Not needed
- Determine if we should create a regression test for this bug: No

**Comment by s77rt** [comment]
@cubuspl42 Sorry I didn't mean to take the C+ role. Just wanted to push this since it was a deploy blocker. Please apply in the Upwork job instead.

**Comment by cubuspl42** [comment]
@s77rt No need; you did the actual job. If it was a deploy blocker, that was the right way to proceed.

**Comment by s77rt** [comment]
Checklist filled already in https://github.com/Expensify/App/issues/28328#issuecomment-1743836303

**Comment by flaviadefaria** [comment]
Switching this to daily so that I can issue payment this week.

**Comment by flaviadefaria** [comment]
Payment Summary:

> For reference, here are some details about the assignees on this issue:
> @s77rt requires payment [offer](https://www.upwork.com/nx/wm/offer/26983371) (Reviewer)
> @redpanda-bit requires payment
> @gadhiyamanan requires payment [offer](https://www.upwork.com/nx/wm/offer/26983373) (Reporter)

@s77rt requires -  $500 - 50%(regression) = $250
@redpanda-bit -  $500 - 50%(regression) = $250
@gadhiyamana - $50 (reporter)

**Comment by s77rt** [comment]
@flaviadefaria This did not cause a regression (it was a regression from another PR). No penalties here.

**Comment by flaviadefaria** [comment]
Thanks @s77rt, here's the updated Payment Summary:

For reference, here are some details about the assignees on this issue:
@s77rt requires payment [offer](https://www.upwork.com/nx/wm/offer/26983371) (Reviewer)
@redpanda-bit requires payment
@gadhiyamanan requires payment [offer](https://www.upwork.com/nx/wm/offer/26983373) (Reporter)

@s77rt requires - $500  
@redpanda-bit - $500 
@gadhiyamana - $50 (reporter)

Issuing payment now.

**Comment by flaviadefaria** [comment]
@s77rt requires - offer sent
@redpanda-bit - please share your UW profile link since I can't find you to send an offer
@gadhiyamana - $50 (reporter) - paid

**Comment by redpanda-bit** [comment]
@flaviadefaria here it is https://www.upwork.com/freelancers/~010e9950f7b5f0f0eb.

**Comment by flaviadefaria** [comment]
@redpanda-bit offer sent!

**Comment by flaviadefaria** [comment]
@s77rt requires - offer sent
@redpanda-bit - paid
@gadhiyamana - paid

@s77rt please accept the offer so that I can pay you and close this GH.

**Comment by s77rt** [comment]
@flaviadefaria No payment is due for me here. Just wanted to resolve the deploy blocker. This can be closed.

**Comment by flaviadefaria** [comment]
Ah, thanks for clarifying - closing!
